File: Introductory_Problems/Weird Algorithm/sol.py
################################################
#                                              #
#   author: raghav_0901 (Raghav Dalmia)        #
#   updated on: 02-10-2020, 10:15:21 pm IST    #
#                                              #
################################################

from math import *

if __name__ == "__main__":
    n = int(input())
    while True:
        print(n, end=" ")
        if(n == 1):
            break
        n = ((n * 3) + 1 if(n % 2 == 1) else n // 2)

File: Introductory_Problems/Weird Algorithm/sol.cpp
/***********************************************
 *                                             *
 *   author: raghav_0901 (Raghav Dalmia)       *
 *   updated on: 02-10-2020, 10:10:53 pm IST   *
 *                                             *
\***********************************************/

#include <bits/stdc++.h>
using namespace std;

int main( )
{
    int n;
    cin >> n;
    while (1)
    {
        cout << n << ' ';
        if (n == 1)
            break;
        n = ((n & 1) ? (n * 3) + 1 : n / 2);
    }
    return 0;
}
